context/src/main/java/io/grpc/PersistentHashArrayMappedTrie.java

```diff
@@ -221,13 +245,21 @@ public V get(K key, int hash, int bitsConsumed) {
             newValues,
             compressedIndex + 1,
             values.length - compressedIndex);
-        return new CompressedIndex<K,V>(newBitmap, newValues);
+        int newSize = size();
+        newSize += 1; // The new leaf
+        for (int i = compressedIndex; i < values.length; i++) {
```

Comment:
I think we just want to `size() + 1` here. The two arraycopy calls ensure all the original items are in the new array.

Reply:
Done.